Issue/21283 no connection message

nbradbury opened the pull request:
Fixes #21283 

Prior to this PR, there were a few places in the app where the user would see both "No network available" and "There is no network available" when there's no network connection. This PR changes this so only the second message appears, and a Retry button was added to the self-hosted user list for consistency.

The following four areas contain this change:

* SelfHostedUserComposables - self-hosted user list
* GifMediaDataSource - adding to media library from Tenor
* StockMediaDataSource - adding to media library from free photo library
* MediaLibraryDataSource - adding media to a post from WP media library

![media-library](https://github.com/user-attachments/assets/7a23ee9f-5a65-4553-923e-4e7b153a211c)

nbradbury committed:
Changed redundant network messaging

nbradbury committed:
Dropped secondary connection message

nbradbury committed:
First pass at Retry button for self-hosted users

nbradbury committed:
Second pass at Retry button for self-hosted users

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

nbradbury committed:
Removed unnecessary box

nbradbury committed:
Fixed tests

dcalhoun reviewed (approved):
Nice improvement. 👏🏻 Thanks for doing this. The following are suggestions we could implement now or later. 

The a11y label for the offline image (struck through cloud) has an odd, likely outdated alt text of "Person reading device with notifications." In fact, I noted several images use this same alt text (e.g., device media permissions illustration). 

I note that when a search field is focused, a screen reader is unable to access the network offline message. One can only move focus to the visible illustration, which includes the aforementioned odd alt text. I.e., when the search field is focused, it is not clear at all that a lack of network connection will disrupt the experience. If one dismisses the virtual keyboard, then you can move screen reader focus to the error message and retry button. This experience is present within the GIF library, free stock media, and editor media insert flows. 

Additionally, I note there are slight divergences for the design of these screens. Some of it likely pre-existed, unsure if the new _Users_ screen differences were intentional. 

| Media Add | Media Insert | Users |
| - | - | - |
| ![android-media-add](https://github.com/user-attachments/assets/ec4401d8-6ca0-43ed-b757-d4b9b3efc1c6) | ![android-media-insert](https://github.com/user-attachments/assets/1561f5ce-e315-4c11-b369-e14b945f7f3a) | ![android-users](https://github.com/user-attachments/assets/591169f8-578d-4b7b-9f23-94ccba9bf415) |

nbradbury committed:
Remove content description from ActionableEmptyView

nbradbury committed:
Remove content description from ActionableEmptyView

nbradbury committed:
Use RoundedCornerShape for retry button

nbradbury commented:
> The a11y label for the offline image (struck through cloud) has an odd, likely outdated alt text of "Person reading device with notifications."

Oh, that's not good. Looks like the pre-existing offline screens all use a custom view (`ActionableEmptyView`) which has that `contentDescription` hard-coded to that string. For this I think the best thing is to remove the `contentDescription` since the image isn't really important.

> I note that when a search field is focused, a screen reader is unable to access the network offline message...

I filed #21288 for this as it's a pre-existing issue.

> I note there are slight divergences for the design of these screens.

Thanks for catching that! I've adjusted the user list offline image size, margin, & tint to more closely match the pre-existing screens. 

The user list Retry button is rounded because that screen uses Material 3 which defaults to [rounded buttons](https://m3.material.io/components/buttons/overview#d3da4750-ed9c-4241-a17d-81c3c5ebb0ca), whereas the other screens use Material 2. I've changed the user list button to match the pre-existing screens
